tests/unit/amo/reducers/test_collections.js

```diff
@@ -1261,4 +1263,79 @@ describe(__filename, () => {
       ).toEqual({ collection_sort: sort, page });
     });
   });
+
+  describe('collectionUrl', () => {
+    it('returns a URL for a collection', () => {
+      const authorUsername = 'some-username';
+      const slug = 'some-slug';
+      const collection = createInternalCollection({
+        detail: createFakeCollectionDetail({ authorUsername, slug }),
+      });
+
+      expect(collectionUrl({ collection })).toEqual(
+        `/collections/${authorUsername}/${slug}/`,
+      );
+    });
+
+    it('returns a URL for an authorUsername / collectionSlug', () => {
+      const authorUsername = 'some-username';
+      const slug = 'some-slug';
+
+      expect(
+        collectionUrl({
+          authorUsername,
+          collection: null,
+          collectionSlug: slug,
+        }),
+      ).toEqual(`/collections/${authorUsername}/${slug}/`);
+    });
+
+    it('returns a URL for the collection when all 3 params are passed', () => {
+      const authorUsername = 'some-username';
+      const slug = 'some-slug';
+      const collection = createInternalCollection({
+        detail: createFakeCollectionDetail({ authorUsername, slug }),
+      });
+
+      expect(
+        collectionUrl({
+          authorUsername: 'a different username',
+          collection,
+          collectionSlug: 'a-different-slug',
+        }),
+      ).toEqual(`/collections/${authorUsername}/${slug}/`);
+    });
+  });
+
+  describe('collectionEditUrl', () => {
+    it('returns an edit URL for a collection', () => {
```

Comment:
Rather than repeat all of the logic you've already tested, I think it's better just to test with dependency injection. This will help keep the tests maintainable as well as enforce a contract between the dependencies.

Example:

```js
const collectionEditUrl = ({
  authorUsername,
  slug,
  collection,
  _collectionUrl = collectionUrl
}) {
  return `${_collectionUrl(...)}edit/`;
}

// After that you really only need one test:

const _collectionUrl = sinon.spy(() => '/base/url/');
expect(collectionEditUrl({ ...params, _collectionUrl })).toEqual(
  '/base/url/edit/',
);
sinon.assert.calledWith(_collectionUrl, params);
```